### PR TITLE
fix: filterbox apply single value

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

filter box can not apply a "single value" filter to the other chats on the dashboard.
related issue: https://github.com/apache/superset/issues/14804

How to reproduce
1. create a filterbox chart and do not choose `Allow multiple selections`
image
2. create a bignumber chart 
3. save big number chart and filterbox in same dashboard

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/2016594/119666349-dfc3eb80-be67-11eb-908b-84257d9cb2bf.mp4

### ADDITIONAL INFORMATION

- [X] Has associated issue: Fixes: https://github.com/apache/superset/issues/14804
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by geido — APPROVED** [review]

File: superset-frontend/src/visualizations/FilterBox/FilterBox.jsx [hunk 1]
```diff
@@ -159,9 +159,10 @@ class FilterBox extends React.PureComponent {
       if (Array.isArray(options)) {
         vals = options.map(opt => (typeof opt === 'string' ? opt : opt.value));
       } else if (options.value) {
-        vals = options.value;
+        // must use array member for legacy extra_filters's value
+        vals = [options.value];
```
[the remark this answers is not in the document]
Comment (villebro) [reply superset-frontend/src/visualizations/FilterBox/FilterBox.jsx:163]: Could we simplify this further by using `ensureIsArray`? Something like

```javascript
...else {
    vals = ensureIsArray(options.value ?? options);
}
```

**Comment by junlincc** [comment]
BEFORE:
https://user-images.githubusercontent.com/67837651/119753997-95853f00-be54-11eb-89bf-5af2a01bdbb9.mov

**Review by junlin-qa — APPROVED** [review]

**Review by villebro — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**